### PR TITLE
[Content] Getting started and Guidelines sections to the new documentation page

### PR DESCRIPTION
**Pull request by ronijaakkola**
## Description
Adds the "Getting started" and "Guidelines" sections and pages to the new documentation site.

## Related Issues
https://helsinkisolutionoffice.atlassian.net/browse/HDS-1184
https://helsinkisolutionoffice.atlassian.net/browse/HDS-1222

## How Has This Been Tested?
Tested by running the documentation site locally.

### CODE REVIEW

File: new-site/gatsby-config.js [hunk 1]
```diff
@@ -10,6 +10,34 @@ module.exports = {
...
+            name: 'Designer',
+            link: '/getting-started/designer',
+            withDivider: true,
+          },
+          {
+            name: 'Developer',
+            link: '/getting-started/developer',
+          },
+          {
+            name: 'FAQ',
```
Comment (ronijaakkola) [inline new-site/gatsby-config.js:31]: @minevala This does not work as expected. I would like to have only the link to the FAQ page and not links to Tab-pages below them. How can I fix this?

**Comment by ronijaakkola** [comment]
SideNavigation would need a max-width or something. Now its width changes depending on the longest menu item label.

**Comment by ronijaakkola** [comment]
We could use a way to control which one of the SideNavigation collapsible links are opened. On some pages it would make sense to open the only collapsible link but on other pages (if there are a multiple of these) closing them on default would be better.

File: new-site/gatsby-config.js (hunk 1 above)
Comment (minevala) [reply new-site/gatsby-config.js:36]: It might try to find subpages with path. I need to check this.

Comment (ronijaakkola) [reply new-site/gatsby-config.js:36]: It is one level higher than our Component pages are. It probably would do the same in Component pages too but the navigation cannot display more levels than 2.

**Comment by minevala** [comment]
> We could use a way to control which one of the SideNavigation collapsible links are opened. On some pages it would make sense to open the only collapsible link but on other pages (if there are a multiple of these) closing them on default would be better.

There is a prop defaultOpenMainLevels for SideNavigation, maybe that could help?

File: new-site/src/docs/guidelines/localisation.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+| Estonian      | `ET`          | <span lang="et">Eesti keelel (ET)</span>            |
+| Spanish       | `ES`          | <span lang="es">En español (ES)</span>              |
+| Northern Sami | `SE`          | <span lang="se_no">Davvisámegillii (SE)</span>      |
+| Inari Sami    | `SMI`         | <span lang="se_fi">Anarâškielân (SMI)</span>        |
+| Skolt Sami    | `SKS`         | <span lang="se_se">Nuõrttsää´mǩiõ´lle (SKS)</span>  |
+| Somali        | `SO`          | <span lang="so">Af soomaali (SO)</span>             |
+| Russian       | `RU`          | <span lang="ru">По-русски (RU)</span>               |
+| Japanese      | `JA`          | <span lang="ja">日本語 (JA)</span>                   |
+| Chinese       | `ZH`          | <span lang="zh">中文</span> (ZH)                     |
+| Arabic        | `AR`          | <span lang="ar" dir="rtl">بالعربية</span> (AR)      |
```
Comment (minevala) [inline new-site/src/docs/guidelines/localisation.mdx:55]: The table is missing the caption

File: new-site/src/docs/guidelines/data-formats.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Time range (in text)                  | 12.30–16.30                            | When presenting times in the written text, use a single dot instead of a colon separator.
+Date range                            | 2.–15. January 2020 / 2.1.–5.2.2020    | Year and/or month can be only written once if both dates share the same year/month.
+
+### Time zones
+**Automatically use the user's timezone if possible.** If not available, default to Finnish time and present the time zone as follows:
+
+Type       | Format                          | Notes
+-----------|---------------------------------|---------------------------------------------------------------------------------------------
+Time zone  | UTC+2                           | Leading or trailing zeroes are not recommended to be used for timezones.
+Time       | 15:00 UTC+2                     | -
```
Comment (minevala) [inline new-site/src/docs/guidelines/data-formats.mdx:42]: The table is missing the caption

File: new-site/src/docs/guidelines/data-formats.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Today, tomorrow, yesterday      | Today, 12:30                  | If talking about today, tomorrow, or yesterday, the written format is used instead of date.
+
+### Ranges
+When presenting time or date ranges, consider making information simpler by leaving redundant month and year information out. **Note that when presenting ranges, en dash (–) should be used instead of a standard hyphen (-).**
+
+Type                                  | Format                                 | Notes
+--------------------------------------|----------------------------------------|---------------------------------------------------------------------------------------
+Time range (forms, schedules, etc.)   | 12:30–16:30                            | When presenting times in forms and schedules, use leading zeros and a colon separator. 
+Time range (in text)                  | 12.30–16.30                            | When presenting times in the written text, use a single dot instead of a colon separator.
+Date range                            | 2.–15. January 2020 / 2.1.–5.2.2020    | Year and/or month can be only written once if both dates share the same year/month.
```
Comment (minevala) [inline new-site/src/docs/guidelines/data-formats.mdx:34]: The table is missing the caption

**Review by minevala — COMMENTED** [review]
Nice additions 👍

File: new-site/src/docs/getting-started/designer.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+title: "Designer"
+---
+
+import { Link } from "hds-react";
+import LeadParagraph from '../../components/LeadParagraph';
+import Image from '../../components/LeadParagraph';
+
+# Getting started as a Designer
+
+<LargeParagraph>
```
[the remark this answers is not in the document]
Comment (ronijaakkola) [reply new-site/src/docs/getting-started/designer.mdx:12]: [x] Fixed LeadParagraphs in the Designer and Developer Getting started pages

File: new-site/src/docs/getting-started/developer.mdx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+slug: "/getting-started/developer"
+title: "Developer"
+---
+
+import { Link } from "hds-react";
+import LeadParagraph from '../../components/LeadParagraph';
+
+# Getting started as a Developer
+
+<LargeParagraph>
```
[the remark this answers is not in the document]
Comment (ronijaakkola) [reply new-site/src/docs/getting-started/developer.mdx:11]: - [x] Fixed LeadParagraphs in the Designer and Developer Getting started pages

File: new-site/src/docs/guidelines/localisation.mdx (hunk 2 above)
Comment (ronijaakkola) [reply new-site/src/docs/guidelines/localisation.mdx:55]: - [x] Added missing table captions

File: new-site/src/docs/guidelines/data-formats.mdx (hunk 3 above)
Comment (ronijaakkola) [reply new-site/src/docs/guidelines/data-formats.mdx:44]: - [x] Added missing table captions

File: new-site/gatsby-config.js (hunk 1 above)
Comment (minevala) [reply new-site/gatsby-config.js:36]: The fragments require:
export default ({ children }) => <>{children}</>;

Comment (ronijaakkola) [reply new-site/gatsby-config.js:36]: > export default ({ children }) => <>{children}</>;

While this indeed makes the pages function as fragments, it does not affect the side navigation at all. Tabs are still part of the navigation but now they open as fragment pages without navigation and footer.

image

image

**Pull request merged**